google/cloud/compute/CMakeLists.txt

```diff
@@ -14,16 +14,78 @@
 # limitations under the License.
 # ~~~
 
-include(GoogleapisConfig)
-set(DOXYGEN_PROJECT_NAME "Cloud Compute API C++ Client")
-set(DOXYGEN_PROJECT_BRIEF "A C++ Client Library for the Cloud Compute API")
-set(DOXYGEN_PROJECT_NUMBER "${PROJECT_VERSION} (Experimental)")
-set(DOXYGEN_EXCLUDE_SYMBOLS "internal")
-set(DOXYGEN_EXAMPLE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/quickstart)
+# Defines a compute proto library and installs the .proto and .pb.h files.
+function (compute_proto_library library)
+    cmake_parse_arguments(_opt "" "" "DEPS;PROTO_FILES" ${ARGN})
+    string(REPLACE "google_cloud_cpp_" "" alias "${library}")
+    google_cloud_cpp_proto_library(
+        "${library}" "${_opt_PROTO_FILES}" PROTO_PATH_DIRECTORIES
+        "${EXTERNAL_GOOGLEAPIS_SOURCE}" "${PROTO_INCLUDE_DIR}"
+        "${PROJECT_SOURCE_DIR}")
+    external_googleapis_set_version_and_alias("${alias}")
+    target_link_libraries("${library}" PUBLIC "${_opt_DEPS}")
+    # Some files are too big for the MSVC defaults.
+    if (MSVC)
+        target_compile_options("${library}" PRIVATE "/bigobj")
+    endif ()
+    google_cloud_cpp_install_proto_library_protos("${library}"
+                                                  "${PROJECT_SOURCE_DIR}")
+    google_cloud_cpp_install_proto_library_headers("${library}")
+endfunction ()
+
+# Defines a compute service library and installs the headers.
+function (compute_service_library library)
+    cmake_parse_arguments(_opt "" "" "DEPS;SRC_FILES" ${ARGN})
+    string(REPLACE "google_cloud_cpp_" "" service_name "${library}")
+    string(REPLACE "_" "-" alias "${service_name}")
+    add_library(${library} ${_opt_SRC_FILES})
```

Comment:
optional: Instead of globbing then passing the source files to this function, we could have this function do the globbing. It would reduce a little bit of duplication between the operation services and the other services.